### PR TITLE
Update label sync to v20171219-22a3ef0f6

### PR DESCRIPTION
**Pull request by fejta**
`v20171215-9c0f9501e` fails with the following:
```
  Warning  Failed                 43s   kubelet, gke-prow-default-pool-116f5d3e-rdf6  Error: failed to start container "label-sync": Error response from daemon: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/app/label_sync/label_sync-image.binary\": stat /app/label_sync/label_sync-image.binary: no such file or directory"
```

It does not appear as though `go_image()` produces valid images.

Pushing a new image now that we bumped rules_docker

/assign @cblecker 
/cc @ixdy @mattmoor

### CODE REVIEW

**Comment by ixdy** [comment]
Why not just roll forward to `rules_docker` HEAD which has a fix for this?

**Comment by fejta** [comment]
Because this works, I see no value from go_image, and I don't understand how we got into a state where this broke in the first place.

**Comment by ixdy** [comment]
The new form is slightly less verbose?
```python
go_image(
  name = "label_sync-image",
  base = "@distroless-base//image",
  binary = ":label_sync"
)
```
@mattmoor could probably explain when we'd use `go_image` vs `container_build` better.

Also, we have a bunch of other `go_image` rules in test-infra, so we probably need to bump rules_docker regardless.

**Comment by fejta** [comment]
Saving a line or two isn't worth me deploying a broken image

**Comment by ixdy** [comment]
`rules_docker` bump in #6001, since we need this anyway.

**Comment by fejta** [comment]
Let me remake the images with the new rules_docker
/hold

**Comment by fejta** [comment]
Updated
/assign @ixdy 
/hold cancel

File: label_sync/cluster/label_sync_job.yaml [hunk 1]
```diff
@@ -26,12 +26,13 @@ spec:
...
       containers:
       - name: label-sync
-        image: gcr.io/k8s-testimages/label_sync:v20171215-9c0f9501e
+        image: gcr.io/k8s-testimages/label_sync:v20171219-22a3ef0f6
         args:
         - --config=/etc/config/labels.yaml
         - --dry-run=false
         - --orgs=kubernetes
         - --token=/etc/github/oauth
+        - --only=kubernetes/contrib
```
Comment (ixdy) [inline label_sync/cluster/label_sync_job.yaml]: why this change?

Comment (fejta) [reply label_sync/cluster/label_sync_job.yaml]: I've been using it as a canary job before the cronjob mutates all repos. Good way to discover problems

Comment (ixdy) [reply label_sync/cluster/label_sync_job.yaml]: maybe add a comment somewhere indicating this is a canary job? it's not very obvious otherwise.

Comment (fejta) [reply label_sync/cluster/label_sync_job.yaml]: meh

Comment (ixdy) [reply label_sync/cluster/label_sync_job.yaml]: as this is it looks like a bug. I would likely look at this later and go "weird, this should probably be updating everything" and remove the line.
